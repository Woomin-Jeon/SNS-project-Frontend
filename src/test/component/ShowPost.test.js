import React from 'react';
import {render} from "enzyme";
import ShowPostOthersPage from "../../components/posts/ShowPost";

describe('ShowPostOthersPage', () => {
  let currentUserState;
  let postState;
  let setPostState;
  let commentState;
  let setCommentState;
  let loginState;
  let setLoginState;
  let p;
  let index;

  beforeEach(() => {
    currentUserState = {
      id: 'Random_ID',
      userName: 'Woomin'
    };
    loginState = { users: [] };
    p = {
      uniqueKey: 999,
      name: 'Woomin',
      time: [2020, 4, 21, 5, 16],
      thumbCount: [],
    };
    commentState = { comment: [] };
  });

  it('renders well', () => {
    const component = render(
      <ShowPostOthersPage
        postState={postState}
        setPostState={setPostState}
        currentUserState={currentUserState}
        commentState={commentState}
        setCommentState={setCommentState}
        p={p}
        index={index}
        loginState={loginState}
        setLoginState={setLoginState}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toMatch('Woomin');
    expect(component.text()).toMatch('좋아요');
    expect(component.text()).toMatch('댓글');
    expect(component.text()).toMatch('스크랩');
    expect(component.text()).toMatch('2020년 4월 21일　5 : 16');
    expect(component.find('.showpost-edit').text()).toBe('수정');
    expect(component.find('.showpost-remove').text()).toBe('삭제');
  });
});
