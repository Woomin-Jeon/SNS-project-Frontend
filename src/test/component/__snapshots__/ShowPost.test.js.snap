// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPostOthersPage renders well 1`] = `ShallowWrapper {}`;
